bin/simbagen.py

```diff
@@ -168,11 +168,13 @@ def __init__(self, filename, endianess):
         self.endianess = endianess
         self.settings = OrderedDict()
 
-        if filename is None:
+        if not filename:
```

Comment:
If `--settings` is given to the script it should be a valid filename. Giving an empty string should not be treated as a filename (even if the file cannot exist).

We should undo this change and make the platformio build script only give `--settings` if a settings file is configured in `platformio.ini`.